client/angular/components/auth/signup_form/auth_signup_form.coffee

```diff
@@ -3,24 +3,34 @@ EventBus    = require 'shared/services/event_bus'
 AuthService = require 'shared/services/auth_service'
 I18n        = require 'shared/services/i18n'
 
+{ hardReload }    = require 'shared/helpers/window'
 { submitOnEnter } = require 'shared/helpers/keyboard'
 
 angular.module('loomioApp').directive 'authSignupForm', ->
   scope: {user: '='}
   templateUrl: 'generated/components/auth/signup_form/auth_signup_form.html'
   controller: ['$scope', ($scope) ->
     $scope.recaptchaKey = AppConfig.recaptchaKey
+    $scope.termsUrl     = AppConfig.theme.terms_url
+    $scope.privacyUrl   = AppConfig.theme.privacy_url
     $scope.name         = $scope.user.name
-    $scope.vars         = {name: $scope.name}
+    $scope.vars         = {name: $scope.name, site_name: AppConfig.theme.site_name}
     $scope.allow        = ->
       AppConfig.features.app.create_user or AppConfig.pendingIdentity.identity_type?
 
     $scope.submit = ->
+      # prevent submit on enter if legal not accepted
+      return if $scope.termsUrl && !$scope.vars.legalAccepted
+
       if $scope.vars.name
         $scope.user.errors = {}
         EventBus.emit $scope, 'processing'
-        $scope.user.name  = $scope.vars.name
-        AuthService.signUp($scope.user).finally -> EventBus.emit $scope, 'doneProcessing'
+        $scope.user.name           = $scope.vars.name
+        $scope.user.legalAccepted  = $scope.vars.legalAccepted
+        AuthService.signUp($scope.user).then (data) ->
```

Comment:
I need a way to tell the client if they are fully signed in (because they have a pending membership or identity with same email address), or if we need to tell them to check their email for a login token. So that's what the returning signed_in is about.